services/github/github-total-star.service.js

```diff
@@ -0,0 +1,259 @@
+    Joi.object({
+      organization: Joi.object({
+        repositories: repositoriesSchema,
+      }).required(),
+    }).required()
+  ).required(),
+}).required()
+
+const userQuery = gql`
+  query fetchStars(
+    $user: String!
+    $nextCursor: String
+    $affiliations: [RepositoryAffiliation]!
+  ) {
+    user(login: $user) {
+      repositories(
+        first: 100
+        after: $nextCursor
+        affiliations: $affiliations
+        orderBy: { field: STARGAZERS, direction: DESC }
+      ) {
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+        nodes {
+          stargazers {
+            totalCount
+          }
+        }
+      }
+    }
+  }
+`
+
+const orgQuery = gql`
+  query fetchStars($user: String!, $nextCursor: String) {
+    organization(login: $user) {
+      repositories(
+        first: 100
+        after: $nextCursor
+        orderBy: { field: STARGAZERS, direction: DESC }
+      ) {
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+        nodes {
+          stargazers {
+            totalCount
+          }
+        }
+      }
+    }
+  }
+`
+const affiliationsAllowedValues = [
+  'OWNER',
+  `COLLABORATOR`,
+  'ORGANIZATION_MEMBER',
+]
+/**
+ * Validates affiliations should contain combination of allowed values in any order.
+ *
+ * @param {string} value affiliation current value
+ * @param {*} helpers object to construct custom error
+ *
+ * @returns {string} valiadtion error or value unchanged
+ */
+const validateAffiliations = (value, helpers) => {
+  const values = value.split(',')
+  if (values.some(e => !affiliationsAllowedValues.includes(e))) {
+    return helpers.error('any.invalid')
+  }
+  return value
+}
+
+const queryParamSchema = Joi.object({
+  org: Joi.valid('').optional(),
+  // TODO add regex
+  affiliations: Joi.string().default('OWNER').custom(validateAffiliations),
+}).required()
+
+module.exports = class TotalStarService extends GithubAuthV4Service {
+  static get defaultLabel() {
+    return 'Total Stars'
+  }
+
+  static get category() {
+    return 'social'
+  }
+
+  static get route() {
+    return {
+      base: 'github/users',
+      pattern: ':user/total-stars',
+      queryParamSchema,
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: "Github User's total stars",
+        namedParams: {
+          user: 'chris48s',
+        },
+        queryParams: { affiliations: 'OWNER,COLLABORATOR' },
+        staticPreview: {
+          label: this.defaultLabel,
+          message: 54,
+          style: 'social',
+        },
+        documentation: userDocumentation,
+      },
+      {
+        title: "Github Org's total stars",
+        namedParams: {
+          user: 'badges',
+        },
+        queryParams: { org: null },
+        staticPreview: {
+          label: this.defaultLabel,
+          message: metric(7000),
+          style: 'social',
+        },
+        documentation: orgDocumentation,
+      },
```

Comment:
This is working for me but it was running on the slow side in the review app. May just be a fluke, but should probably poke around a bit to make sure this badge is providing quickly enough to meet the github/camo window

https://shields-staging-pr-5507.herokuapp.com/github/users/badges/total-stars?org&style=social
![](https://shields-staging-pr-5507.herokuapp.com/github/users/badges/total-stars?org&style=social)

Reply:
did no changes for this, but verified only two sequential call is getting fired.